base/samples/LaserScan.hpp

```diff
@@ -130,6 +132,53 @@ namespace base { namespace samples {
 		}
 	    }
 	}
+	
+	/**Converts the laser scan into an interpolated pointcloud according to the given transformation, 
+	 * the startTime, the angular_resolution and the rotation speed of the laserbeam.
+	 * Acts just like convertScanToPointCloud() but considers the motion of the laser scanner.
+	 */
+	template<typename T, typename Trans>
+    void convertScanToPointCloudInterpolated(std::vector<T> &points,
+                                                const Trans& transformation,
+                                                const base::Time startTime,
+                                                bool skip_invalid_points = true)const
+    {
+        points.clear();
+        
+        //give the vector a hint about the size it might be
+        points.reserve(ranges.size());
+
+        //this is optimized for runtime
+        //moving the check for skip_invalid_points
+        //out of the loop speeds up the execution 
+        //time by ~25 %
+        if(!skip_invalid_points)
+        {
+            for(unsigned int i = 0; i < ranges.size(); i++) {
+                base::samples::RigidBodyState bodyState;
+                Eigen::Vector3d point;
+                if(getPointFromScanBeamXForward(i, point) && transformation.get(startTime + base::Time::fromSeconds((start_angle / angular_resolution + i) * 
+                                                                                (angular_resolution / speed)), bodyState, false)) 
```

Comment:
This looks weird ... You call this "Interpolated" but are not interpolating the transform. I guess that you should make this a parameter of the method.